payload_stage2/source/main.c

```diff
@@ -13,9 +13,19 @@
 extern u8 screen_init_bin[];
 extern u32 screen_init_bin_size;
 
+static inline void* copy_memory(void *dst, void *src, size_t amount)
```

Comment:
As a heads up, PIC code isn't compatible with the newlib standard library shipped with devKitARM, so I included a simple replacement for memcpy here.